src/builtins.ts

```diff
@@ -658,6 +661,34 @@ export function compileCall(
       if (!type) return module.unreachable();
       return module.i32(type.isManaged ? 1 : 0);
     }
+    case BuiltinSymbols.isVoid: { // isVoid<T>() -> bool
+      let type = evaluateConstantType(compiler, typeArguments, operands, reportNode);
+      compiler.currentType = Type.bool;
+      if (!type) return module.unreachable();
+      return module.i32(type.kind === TypeKind.VOID ? 1 : 0);
+    }
+    case BuiltinSymbols.lengthof: { // lengthof<T>(): i32
+      let type = evaluateConstantType(compiler, typeArguments, operands, reportNode);
+      compiler.currentType = Type.i32;
+      if (type === null) {
+        compiler.error(
+          DiagnosticCode.Expected_0_type_arguments_but_got_1,
+          reportNode.range, "1", (typeArguments ? typeArguments.length : 1).toString(10)
+        );
+        return module.unreachable();
+      }
+
+      if (type.signatureReference === null) {
+        compiler.error(
+          DiagnosticCode.Type_0_has_no_call_signatures,
+          reportNode.range, "1", (typeArguments ? typeArguments.length : 1).toString(10)
+        );
+        return module.unreachable();
+      }
+      return type.signatureReference.parameterNames === null
```

Comment:
Perhaps cache `type.signatureReference` and `signatureReference.parameterNames` in locals.

Reply:
Okay. Looks good.